tools/README.md

```diff
@@ -56,4 +58,77 @@ To run nGraph in Docker, choose one of two ways to create your container:
 - After the build completes, you will be able to use the virtualenv located at `/workspace/build_cmake/venv-tf-py3` and run a test.
 
         source build_cmake/venv-tf-py3/bin/activate
-        python examples/infer_image.py
\ No newline at end of file
+        python examples/infer_image.py
+
+
+
+## Introducing grappler and tf2ngraph
+
+#### Normal mode of operations
+The normal mode of operation of `ngrph_bridge` is:
+```
+import ngraph_bridge
+
+in0, in1, out0 = construct_network()
+sess = tf.Session()
+sess.run(out0, feed_dict = {in0:[2,3], in1:[4,5]})
+```
+
+The "normal" mode latches onto `Tensorflow` operations by registering `GraphOptimizationPass`es such as `NGraphRewritePass` and `NGraphVariableCapturePass`. It also registers a custom op `NGraphEncapsulate`. Note that the graph rewriting and the actual execution all happens when `session.run` is called.
+
+
+#### Introducing `Grappler`
+`Grappler` can be thought of as a function that accepts a `graphdef` and returns a (most likely) new modified `graphdef`, although in cases of failure to transform, `Tensorflow` will silently continue to run with the input untransformed `graphdef`. Recently we have added a way to build `ngraph-bridge` with `Grappler`, by using the `--use_grappler_optimizer` flag in `build_ngtf.py`. We register the `NgraphOptimizer` as our `Grappler` pass. It pretty much does the same rewriting that `NGraphRewritePass` and `NGraphVariableCapturePass` was doing earlier, except for some subtle differences. For example, when `grappler` receives the graph in a certain stage of the `Tensorflow` execution pipeline which is different from when the `GraphOptimizationPass`es worked. Also we add `IdentityN` nodes to fetch (outputs), feed (inputs), init and keep ops to ensure we also capture these nodes(if supported), because by default `grappler` leaves them out.
+
+A sample script will look like:
+```
+import ngraph_bridge
+
+in0, in1, out0 = construct_network()
+sess = tf.Session(config=ngraph_bridge.update_config(tf.ConfigProto()))
+result = sess.run(out0, feed_dict = {in0:[2,3], in1:[4,5]})
+```
+
+Notice the new line introduced here wrt the "normal" path. `config_updated = ngraph_bridge.update_config(config)` must now be passed during session construction to ensure `NgraphOptimizer` `grappler` pass is enabled. Like the "normal" path, in this script too the graph rewriting and actual execution happen when `session.run` is called.
```

Comment:
we can add a short description about what the update_config does

Reply:
Added